nuxt/server/api/auth-proxy.ts

```diff
@@ -0,0 +1,60 @@
+import { H3Event, readBody } from 'h3'
+import consola from 'consola'
+import { TURNSTILE_HEADER_KEY } from '~/utils/constants'
+
+export default defineEventHandler(async function (event: H3Event) {
+  const { req, res } = event.node
+  consola.debug(getMethod(event))
+  consola.debug('Request: ' + req)
+  let body: any
+  try {
+    body = await readBody(event)
+  } catch (e) {
+    consola.error(e)
+  }
+
+  switch (body.operationName) {
+    case 'authenticate':
+    case 'accountRegistration':
+      {
+        const turnstileKey = req.headers[TURNSTILE_HEADER_KEY.toLowerCase()]
+        if (Array.isArray(turnstileKey)) {
+          throw createError({
+            statusCode: 422,
+            statusMessage: 'TurnstileKey cannot be an array.',
+          })
+        }
+        if (turnstileKey === undefined) {
+          consola.error('TurnstileKey undefined')
+          throw createError({
```

Comment:
This command pair could be extracted to a helper function

Reply:
I agree. I extracted the code in [this commit](https://github.com/maevsi/maevsi/commit/3641fa38dde20b431936c0d043caa4a68f37f1ba)